test_conformance/api/test_api_consistency.cpp

```diff
@@ -1223,11 +1243,27 @@ int test_consistency_requirements_fp16(cl_device_id deviceID,
 
     if (is_extension_available(deviceID, "cl_khr_fp16"))
     {
+        const Version version = get_device_cl_version(deviceID);
+
         error = clGetDeviceInfo(deviceID, CL_DEVICE_HALF_FP_CONFIG,
                                 sizeof(value), &value, nullptr);
         test_error(error, "Unable to get device CL_DEVICE_HALF_FP_CONFIG");
         test_assert_error(value > 0,
                           "CL_DEVICE_HALF_FP_CONFIG must return nonzero value");
+        if (version < Version(2, 0))
+        {
+            test_assert_error(
+                value & (CL_FP_ROUND_TO_NEAREST | CL_FP_INF_NAN),
```

Comment:
The [specification](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_HALF_FP_CONFIG) for `CL_DEVICE_HALF_FP_CONFIG` doesn't explicitly mention that this config is tied to versions < 2.0:

> the minimum half-precision floating-point capability for OpenCL 2.0 or newer devices is:
> 
> [CL_FP_ROUND_TO_ZERO](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_FP_ROUND_TO_ZERO)
> 
> or
> 
> [CL_FP_ROUND_TO_NEAREST](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_FP_ROUND_TO_NEAREST) |
> [CL_FP_INF_NAN](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_FP_INF_NAN).
> 

  The specification for `..._DOUBLE_FP_CONFIG` is unambiguous, but for `..._HALF_FP_CONFIG` the wording could also be interpreted as "either of these configs can be considered a minimum config for OpenCL 2.0 or newer".

I'm not quite sure which interpretation was intended, maybe @bashbaug can advise?

Reply:
Yeah, I think there's a minor spec bug here, and I just filed https://github.com/KhronosGroup/OpenCL-Docs/issues/1267 to fix it.

My understanding is:

* The OpenCL version supported by the device is irrelevant.
* Devices supporting half-precision (via `cl_khr_fp16`) must either support `CL_FP_ROUND_TO_ZERO` or the combination of `CL_FP_ROUND_TO_NEAREST | CL_FP_INF_NAN`.

This is consistent with the `cl_khr_fp16` description in the last standalone extension doc v3.0.15.